espflash/src/flasher/mod.rs

```diff
@@ -771,6 +771,63 @@ impl Flasher {
         Ok(())
     }
 
+    /// For Viam, load an ELF image to flash and execute it with NVS data included
+    pub fn load_elf_to_flash_with_format_with_nvs(
```

Comment:
Can we instead modify `load_elf_to_flash_with_format()` and `get_flash_image()` to take a `nvs_data: Option<Vec<u8>>` parameter to reduce the duplication?